### PR TITLE
PR - Find Max implementation

### PR DESCRIPTION
**Pull request by msaf9**
Find Max implementation

### CODE REVIEW

**Review by raklaptudirm — DISMISSED** [review]

File: en/Basic Math/Find Max.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+- Makes our process of finding a particular item easy.
+- If we are able to find out and sort our records based on 
+  largest value then we can sort or find our records according 
+  to particular index or element with ease.
+
+## Steps
+
+The maximum number from an array can be found by various ways. 
+Here, we will see three different ways of finding the largest 
+number.
```
Comment (Panquesito7) [inline en/Basic Math/Find Max.md:21]:
```suggestion
number.

```

File: en/Basic Math/Find Max.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+## Example 1
+
+By comparing all the elements of an array:
+
+Given the list `[6, 4, 50, 8, 70, 2, 20]`, let's find the maximum number.
+
+### Step 1
+
+Provide `[6, 4, 50, 8, 70, 2, 20]` as input for a method/ function.
```
Comment (Panquesito7) [inline en/Basic Math/Find Max.md:34]:
```suggestion
Provide `[6, 4, 50, 8, 70, 2, 20]` as input for a method or function.
```

File: en/Basic Math/Find Max.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+Get the last value out of the list by using predefined last method.
+`70`
+
+### Step 4
+
+Return the value 70.
+
+## Implementation
+
+-   [Ruby](https://github.com/TheAlgorithms/Ruby/blob/master/maths/find_max.rb)
```
Comment (Panquesito7) [inline en/Basic Math/Find Max.md:102]:
```suggestion
- [Ruby](https://github.com/TheAlgorithms/Ruby/blob/master/maths/find_max.rb)
```

File: en/Basic Math/Find Max.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+# Source
+
+-   [GeeksForGeeks](https://www.geeksforgeeks.org/c-program-find-largest-element-array/)
```
Comment (Panquesito7) [inline en/Basic Math/Find Max.md:106]:
```suggestion
- [GeeksForGeeks](https://www.geeksforgeeks.org/c-program-find-largest-element-array/)
```

File: en/Basic Math/Find Max.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+# YouTube
+
+-   [Video URL](https://youtu.be/En68ipRaFOU)
```
Comment (Panquesito7) [inline en/Basic Math/Find Max.md:110]:
```suggestion
- [YouTube URL video](https://youtu.be/En68ipRaFOU)
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Good work! 👍 
I would only suggest changing the filename to `Find the maximum number.md`.

File: en/Basic Math/Find Max.md (hunk 5 above)
Comment (msaf9) [reply en/Basic Math/Find Max.md]: I have made all the changes. Thank you for your suggestions.

File: en/Basic Math/Find the maximum number.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
+# Find (Maximum) 
+
+Find the maximum number from a list of given numbers
```
Comment (Panquesito7) [inline en/Basic Math/Find the maximum number.md:3]:
```suggestion
Find the maximum number from a list of given numbers.
```

File: en/Basic Math/Find the maximum number.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+## Applications
+
+By finding the maximum number in a list we can perform many 
+operations based on this.
```
Comment (Panquesito7) [inline en/Basic Math/Find the maximum number.md:8]:
```suggestion
By finding the maximum number in a list, we can perform many operations based on this.
```

File: en/Basic Math/Find the maximum number.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+Uses:
```
Comment (Panquesito7) [inline en/Basic Math/Find the maximum number.md:10]:
```suggestion
Uses:

```

File: en/Basic Math/Find the maximum number.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+Get the last value out of the list by using predefined last method.
+`70`
+
+### Step 4
+
+Return the value 70.
+
+## Implementation
+
+-  [Ruby](https://github.com/TheAlgorithms/Ruby/blob/master/maths/find_max.rb)
```
Comment (Panquesito7) [inline en/Basic Math/Find the maximum number.md:103]:
```suggestion
- [Ruby](https://github.com/TheAlgorithms/Ruby/blob/master/maths/find_max.rb)
```

File: en/Basic Math/Find the maximum number.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+# Source
+
+-  [GeeksForGeeks](https://www.geeksforgeeks.org/c-program-find-largest-element-array/)
```
Comment (Panquesito7) [inline en/Basic Math/Find the maximum number.md:107]:
```suggestion
- [GeeksForGeeks](https://www.geeksforgeeks.org/c-program-find-largest-element-array/)
```

File: en/Basic Math/Find the maximum number.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+# YouTube
+
+-  [YouTube URL video](https://youtu.be/En68ipRaFOU)
```
Comment (Panquesito7) [inline en/Basic Math/Find the maximum number.md:111]:
```suggestion
- [YouTube URL video](https://youtu.be/En68ipRaFOU)
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Almost there! 😄

File: en/Basic Math/Find the maximum number.md (hunk 11 above)
Comment (msaf9) [reply en/Basic Math/Find the maximum number.md]: Yeah, I have implemented all the changes. Thank you @Panquesito7

**Review by Panquesito7 — APPROVED** [review]
LGTM. Thank you for your contribution! 👍 🎉
@appgurueu, would you like to review this PR? Thanks.

After this PR is merged, you can ask for the `Contributor` role in our [Discord server](https://discord.gg/c7MnfGFGa6) in the `#privs-role-requests` channel. 🙂

**Comment by appgurueu** [comment]
Why would you sort an array/list just to find it's max? Finding a max (or min) is plain O(n), sorting a list is O(n log n). I also don't like how this is array-specific. Just looping over all elements works for all collections.

**Comment by vbrazo** [comment]
@appgurueu I think he was just trying to show that there are multiple approaches that are great for beginners. As you described there is a time complexity difference and therefore the `O(n)` approach is more suitable for this context.

**Review by vbrazo — APPROVED** [review]
LGTM

**Pull request merged**